File: URI/matematica/1202.py
def mod_exp(a, b, m):
    if b == 0:
        return 1
    c = mod_exp(a, b // 2, m)
    c = (c * c) % m
    if b % 2 != 0:
        c = (c * a) % m
    return c
    
def func():
    s = input()
    num_mod = 0
    l = int(len(s)) - 1
    exp = 0
    for i in range(l, -1, -1):
        if int(s[i]) == 1:
            num_mod += (num_mod % 1500 +  ((int(s[i]))*mod_exp(2, exp , 1500)))%1500  
        exp += 1  
    print("%03d" % fib[num_mod])
    
n = int(input())

fib = [0, 1, 1]

for i in range(3, 1501):
    fib.append((fib[i - 1] % 1000 + fib[i - 2] % 1000) % 1000) 

while n > 0:
    n -= 1
    func()

File: URI/matematica/1202.cpp
#include <iostream>
#include <cstring>
#include <string>
#define MOD 1500
using namespace std;

typedef long long ll;
int fib[1501];
char s[10010];

void calc(){
    fib[1] = fib[2] = 1;
    for (int i = 3; i <= 1500; i++) {fib[i] = (fib[i-1] % 1000 + fib[i-2] % 1000) % 1000; }
}

int mod_exp(int a, int b){
    if (b == 0) return 1;
    ll c = mod_exp(a, b / 2);
    c = (c * c) % MOD;
    if (b % 2)  c = (c * a) % MOD;
    return c;
}

int main(){
    int n;
    calc();
    scanf("%d", &n);
    while(n--){
        scanf("%s", s);
        int num_mod = 0; int tam = strlen(s);int exp = 0;
        for (int i = tam - 1; i >= 0; i--){
            if(s[i] == '1')
                num_mod = (num_mod + (s[i] - '0') * mod_exp(2, exp)) % MOD;
                
            exp++;
        }
        printf("%03d\n", fib[num_mod]);
    }
    return 0;
}
